src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -101,10 +101,13 @@ private void parseCommit(List<String> lines) {
             if( line.length() < 1)
                 continue;
             if (line.startsWith("commit ")) {
-                this.id = line.split(" ")[1];
+                String[] split = line.split(" ");
+                if (split.length > 0) this.id = split[1];
+                else throw new IllegalArgumentException("Commit has no ID" + lines);
```

Comment:
I would print a warning to the log and then call continue.
Otherwise it makes sense to document the possible <code> IllegalArgumentException</code>